src/node/interfaces.cpp

```diff
@@ -838,6 +839,51 @@ class ChainImpl : public Chain
     NodeContext& m_node;
 };
 
+class BlockTemplateImpl : public BlockTemplate
+{
+public:
+    explicit BlockTemplateImpl(std::unique_ptr<CBlockTemplate> block_template) : m_block_template(std::move(block_template)) {
+        assert(m_block_template);
+    }
+
+    CBlockHeader getBlockHeader() override
+    {
+        return m_block_template->block;
+    }
+
+    CBlock getBlock() override
+    {
+        return m_block_template->block;
+    }
+
+    std::vector<CAmount> getTxFees() override
+    {
+        return m_block_template->vTxFees;
+    }
+
+    std::vector<int64_t> getTxSigops() override
+    {
+        return m_block_template->vTxSigOpsCost;
+    }
+
+    CTransactionRef getCoinbaseTx() override
+    {
+        return m_block_template->block.vtx[0];
+    }
+
+    std::vector<unsigned char> getCoinbaseCommitment() override
+    {
+        return m_block_template->vchCoinbaseCommitment;
+    }
+
+    int getWitnessCommitmentIndex() override
+    {
+        return GetWitnessCommitmentIndex(m_block_template->block);
+    }
+
+    const std::unique_ptr<CBlockTemplate> m_block_template;
```

Comment:
> Does this need to be public?

Convention in `interface.cpp` files is to make class members public since the classes are private and meant to be a minimal as possible (ideally just containing accessors and forwarding methods). Comment at top of the file mentions this:

https://github.com/bitcoin/bitcoin/blob/5d280130446d57d653c749005a2e363265d87686/src/node/interfaces.cpp#L82-L83